src/potentials.h

```diff
@@ -277,21 +430,21 @@ class Hertz : public PairPotentialBase {
  * when r < sum of radii + squarewell_threshold, and zero otherwise.
  *
  */
-class SquareWell : public PairPotentialBase {
+class SquareWell : public MixerPairPotentialBase {
   protected:
-    std::shared_ptr<ParametersTable> m; // table w. squarewell_threshold_ij and squarewell_depth_ij
+    TPairMatrixPtr diameter_sw_squared; // 0.25 * (sigma_i + 2*threshold_i + sigma_j + 2*threshold_j)^2
+    TPairMatrixPtr depth_sw; // geometric mean of epsilon_i and epsilon_j
+    void initPairMatrices() override;
+
   public:
-    SquareWell(const std::string &name = "squarewell");
+    SquareWell(const std::string &name = "squarewell")
+        : MixerPairPotentialBase(name) {};
     inline double operator()(const Particle &a, const Particle &b, const Point &r) const override {
-        if (r.squaredNorm() < m->th(a.id, b.id)) // threshold squared
-            return -m->esw(a.id, b.id);
+        if (r.squaredNorm() < (*diameter_sw_squared)(a.id, b.id))
+            return -(*depth_sw)(a.id, b.id);
```

Comment:
`depth_sw` --> `epsilon_sw`?

Reply:
The resolution of the Hertz potential is already prepared in the next (queued) pull request.